pkg/resources/user_public_keys.go

```diff
@@ -0,0 +1,182 @@
+package resources
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var userPublicKeyProperties = []string{
+	"rsa_public_key",
+	"rsa_public_key_2",
+}
+
+var userPublicKeysSchema = map[string]*schema.Schema{
+	"name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "Name of the user.",
+		ForceNew:    true,
+	},
+	"rsa_public_key": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies the user’s RSA public key; used for key-pair authentication. Must be on 1 line without header and trailer.",
+	},
+	"rsa_public_key_2": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies the user’s second RSA public key; used to rotate the public and Public keys for key-pair authentication based on an expiration schedule set by your organization. Must be on 1 line without header and trailer.",
+	},
+}
+
+func UserPublicKeys() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateUserPublicKeys,
+		Read:   ReadUserPublicKeys,
+		Update: UpdateUserPublicKeys,
+		Delete: DeleteUserPublicKeys,
+
+		Schema: userPublicKeysSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+func ReadUserPublicKeys(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	id := d.Id()
+
+	// First check if user exists
+	stmt := snowflake.User(id).Show()
+	row := snowflake.QueryRow(db, stmt)
+	_, err := snowflake.ScanUser(row)
+	if err == sql.ErrNoRows {
+		// If not found, mark resource to be removed from statefile during apply or refresh
+		log.Printf("[DEBUG] user (%s) not found", d.Id())
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// at this point, we know we have a user. Read keys
+	var rsaKeyFP string
+	var rsaKey2FP string
+
+	stmt = fmt.Sprintf(`DESCRIBE USER "%s"`, d.Get("name").(string))
+	props, err := snowflake.Query(db, stmt)
+	if err != nil {
+		return err
+	}
+	defer props.Close()
+
+	for props.Next() {
+		prop := &snowflake.DescribeUserProp{}
+		err := props.StructScan(prop)
+		if err != nil {
+			return err
+		}
+
+		switch prop.Property {
+		case "RSA_PUBLIC_KEY_FP":
+			if prop.Value.Valid {
+				rsaKeyFP = prop.Value.String
+			}
+		case "RSA_PUBLIC_KEY_2_FP":
+			if prop.Value.Valid {
+				rsaKey2FP = prop.Value.String
+			}
+		default:
+			log.Printf("[DEBUG] skipping user property %s", prop.Property)
+		}
+	}
+
+	d.Set("rsa_public_key_fp", rsaKeyFP)
```

Comment:
🚫 **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Error return value of `d.Set` is not checked (errcheck)